projects/vdk-core/src/vdk/api/job_input.py

```diff
@@ -361,3 +361,18 @@ def get_execution_properties(self) -> dict:
                 SELECT id, {pa__job_start_ts_expr} FROM mytable
         """
         pass
+
+    @abstractmethod
+    def cancel_job_execution(self) -> None:
```

Comment:
We should not that in case of a template, it will cancel the template being executed but the parent job (or template) will not be canceled).